src/pages/HistoryPage/HistoryPage.tsx

```diff
@@ -1,260 +1,451 @@
+        <Text bold css={{ fontSize: '$6' }}>
+          {totalAllocated} {tokenName}
+        </Text>
+      </Panel>
+      {shortPanelShow ? (
+        <Panel
+          nested
+          css={{
+            display: 'flex',
+            flexDirection: 'row',
+            justifyContent: 'flex-end',
+            color: '$primary',
+          }}
+        >
+          <Box css={{ mr: '$md' }}>
+            <Text variant="formLabel">Notes Left</Text>
+            <Text
+              bold
+              css={{
+                fontSize: '$6',
+              }}
+            >
+              {sent.filter(sent => sent.note).length}
+            </Text>
+          </Box>
+          <Box>
+            <Text variant="formLabel">Received</Text>
+            <Text
+              bold
+              css={{
+                fontSize: '$6',
+              }}
+            >
+              {received.filter(received => received.note).length}
+            </Text>
+          </Box>
+        </Panel>
+      ) : (
+        <Panel nested>
+          <Box
+            css={{
+              display: 'flex',
+              gap: '$sm',
+            }}
+          >
+            <Text
+              variant="formLabel"
+              css={{
+                color: tab === 0 ? '$primary' : '$gray400',
+                cursor: 'pointer',
+                border: '1px solid $primary',
+                padding: '$sm',
+                borderRadius: '$pill',
+              }}
+              onClick={() => setTab(0)}
+            >
+              Received
+            </Text>
+            <Text
+              variant="formLabel"
+              css={{
+                color: tab === 1 ? '$primary' : '$gray400',
+                cursor: 'pointer',
+                border: '1px solid $primary',
+                padding: '$sm',
+                borderRadius: '$pill',
+              }}
+              onClick={() => setTab(1)}
+            >
+              Sent
+            </Text>
+          </Box>
+          {tab === 0 ? (
+            received.length > 0 ? (
+              received.map(gift => (
+                <Box key={gift.id} css={{ display: 'flex', my: '$sm' }}>
+                  <Box css={{ mr: '$md' }}>
+                    <ApeAvatar user={gift.sender} />
+                  </Box>
+                  <Box
+                    css={
+                      !gift.note
+                        ? { alignItems: 'center', display: 'flex' }
+                        : {}
+                    }
+                  >
+                    {gift.note && (
+                      <Text css={{ mb: '$xs', lineHeight: 'normal' }}>
+                        {gift.note}
+                      </Text>
+                    )}
+                    <Box css={{ fontSize: '$3', color: '$green' }}>
+                      {gift.tokens} {tokenName} received from {gift.sender.name}
+                    </Box>
+                  </Box>
+                </Box>
+              ))
```

Comment:
@MarcogodoyV Please extract this into its own component so it can be reused for sent gifts/notes as well. Change the text to `{gift.tokens} {tokenName} sent to {gift.sender.name}` in that case.

To test this, it's probably easier to put in some dummy data for `received` and `sent` in this component rather than having to go through the whole allocation process.